packages/svelte/src/internal/client/hydration.js

```diff
@@ -66,9 +78,10 @@ export function get_hydration_fragment(node, insert_text = false) {
  * @returns {void}
  */
 export function hydrate_block_anchor(anchor_node, is_controlled) {
-	/** @type {Node} */
-	let target_node = anchor_node;
-	if (current_hydration_fragment !== null) {
+	if (hydrating) {
```

Comment:
do we need this (and similar) guards? surely we're already only calling functions like `hydrate_block_anchor` when we're hydrating. rollup is smart enough to deal with it

Reply:
Depends on the method, in this case there's no if block at the call site because it would result in more if block code overall.

Reply:
This is what I was talking about last week in our call. I wasn't seeing the tree shaking working as expected by Rollup. However, if you put into this boolean guard then it works great – something I didn't do.